src/PrivateDns/PrivateDns/Records/NewAzurePrivateDnsRecordSet.cs

```diff
@@ -0,0 +1,145 @@
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the records in this record set (relative to the name of the zone and without a terminating dot).")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The type of Private DNS records in this record set.")]
+        [ValidateNotNullOrEmpty]
+        public RecordType RecordType { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The TTL value of all the records in this record set.")]
+        [ValidateNotNullOrEmpty]
+        public uint Ttl { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "A hash table which represents resource tags.")]
+        public Hashtable Metadata { get; set; }
+
+        [Alias("PrivateDnsRecords")]
+        [Parameter(Mandatory = false, HelpMessage = "The private dns records that are part of this record set.")]
+        [ValidateNotNull]
+        public PrivateDnsRecordBase[] PrivateDnsRecord { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Do not fail if the record set already exists.")]
+        public SwitchParameter Overwrite { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string zoneName = null;
+            string resourceGroupName = null;
+            PSPrivateDnsRecordSet result = null;
+
+            if (RecordType == RecordType.SOA)
+            {
+                throw new System.ArgumentException(ProjectResources.Error_AddRecordSOA);
+            }
+
+            switch (this.ParameterSetName)
+            {
+                case FieldsParameterSetName:
+                    zoneName = this.ZoneName;
+                    resourceGroupName = this.ResourceGroupName;
+                    break;
+                case ObjectParameterSetName:
+                    zoneName = this.Zone.Name;
+                    resourceGroupName = this.Zone.ResourceGroupName;
+                    break;
+                case ResourceParameterSetName:
+                    PrivateDnsUtils.GetResourceGroupNameAndZoneNameFromResourceId(this.ParentResourceId, out resourceGroupName, out zoneName);
+                    break;
+            }
+
+            if(zoneName != null && this.Name.EndsWith(zoneName))
+            {
+                this.WriteWarning(string.Format(ProjectResources.Error_RecordSetNameEndsWithZoneName, this.Name, zoneName));
+            }
+
+            if (zoneName != null && zoneName.EndsWith("."))
+            {
+                zoneName = zoneName.TrimEnd('.');
+                this.WriteWarning($"Modifying zone name to remove terminating '.'.  Zone name used is \"{zoneName}\".");
+            }
+
+            if (this.PrivateDnsRecord == null)
+            {
+                this.WriteWarning(ProjectResources.Warning_PrivateDnsRecordsParamNeedsToBeSpecified);
```

Comment:
Why wouldn't you just make PrivateDnsRecord mandatory?

Reply:
Creating empty recordsets should be allowed, this just shows a warning to the user that they have created an empty recordset.